documentation/user-consent.md

```diff
@@ -0,0 +1,220 @@
+      - [JWT-bearer flow](#jwt-bearer-flow)
+      - [CIBA flow](#ciba-flow)
+      - [Client Credentials flow](#client-credentials-flow)
+      - [... (other grant types)](#-other-grant-types)
+  - [Consent API](#consent-api)
+  - [Annex: Flow source code](#annex-flow-source-code)
+
+
+## Introduction
+
+Operator platform implementing CAMARA should be built with a privacy-by-design approach to fully comply with data protection regulations like the [GDPR regulation](https://gdpr-info.eu/) in Europe, which provides a great level of protection for user's privacy. This means that an API that processes Personal Information needs a user consent. Consents are given by users to legal entities to process personal data under a specific purpose.
+
+This document captures guidelines for Operator platform to handle user consents to comply with GDPR or equivalent requirements in an easy way, introducing the concept of purpose in OpenID Connect. Even being defined based on concepts that maps to GDPR regulation, proposed solution and concepts are generic enough to be used by Operators on any country.
+
+The document details aspects regarding the Consent Management, which includes following concepts:
+- CRUD operations for a Consent, i.e.: Create, Read, Update or Delete a Consent Record. Specific details will be handled as for example in fact Consents may not be deleted but Revoked instead.
+- What information is relevant in the managed Consent record, e.g.: end User Identifier to which the Consent is associated, expiration/validity, etc.
+- Policy enforcement to validate existence and validity of Consent before Authorizing access to a (set of) scopes.
+- Allow end Users to view and update their Consents.
+- Flows detailing how Consent can be collected from the User without degrading the user experience while using a third-party service in an Application.
+
+>[TO BE EDITED/COMPLETED]
+
+## Glossary of Terms and Concepts
+
+The list below introduces several key concepts:
+
+-	`Application`: client system that requires access to protected resources. Application must use the appropriate access token to access those resources (e.g. CAMARA Network APIs).
+-	`User`: the human participant which is identified in Telco Operator by a unique user identifier (e.g. Subject identifier sub in OpenID Connect terminology). The user is the resource owner.
+-	`Auth Server`: authorization server which receives requests from applications to issue an access token upon successful authentication and consent of the user. The OpenID Connect Provider is able to authenticate the user validating user identity against the corresponding Identity Provider. The authorization server exposes two endpoints: the Authorization endpoint and the Token endpoint.
+-	`Identity Provider (IdP)`: It corresponds to the OpenID identity provider which is the party that provides user authentication as a service (it creates, maintains, and manages user identity information). 
```

Comment:
This doesn't need to be OpenID Connect does it? Again, let's remove technology from the definition.

Reply:
See https://github.com/camaraproject/IdentityAndConsentManagement/pull/13#discussion_r1195033670